app/controllers/spree/admin/payment_methods_controller_decorator.rb

```diff
@@ -74,7 +74,7 @@ def show_stripe?
 
       def restrict_stripe_account_change
         return unless @payment_method.try(:type) == "Spree::Gateway::StripeConnect"
-        return unless @payment_method.preferred_enterprise_id
+        return unless @payment_method.preferred_enterprise_id.andand > 0
```

Comment:
What about checking there's an actual `@payment_method` first? this way you can avoid lettering the rest of the method with conditionals like `try` or `andand`. We should even raise an error and show it in the UI accordingly when there is no `@payment_method` on an update, but I don't know precisely how this controller works.

Reply:
👍  sounds good

Reply:
Resolved